### PR TITLE
Add online notification switch to settings

### PR DESCRIPTION
**Pull request by kunall17**

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/api/toggleMobilePushSettings.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  {
+    auth,
+    offline,
+    online,
+  }: { auth: Auth, offline?: boolean, online?: boolean, }) =>
+  apiPatch(
+    auth,
+    'settings/notifications',
+    res => res,
+    JSON.parse(JSON.stringify({ // To remove undefined values
```
Comment (borisyankov) [inline src/api/toggleMobilePushSettings.js]: Extract that to an utility function that is called `removeUndefinedValues` then you don't need a comment.

File: src/common/ZulipSwitch.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ export default class ZulipSwitch extends React.PureComponent {
   props: {
     value?: boolean,
     defaultValue: boolean,
-    onValueChange: (arg: boolean) => void,
+    onValueChange: (arg: boolean) => void | Promise<void>,
```
Comment (borisyankov) [inline src/common/ZulipSwitch.js]: This might be better not to change...

File: src/settings/SettingsScreen.js [hunk 3]
```diff
@@ -33,7 +33,8 @@ class SettingsScreen extends React.Component {
 
   props: {
     auth: Auth,
-    notifications: boolean,
+    offlinePushNotification: boolean,
```
Comment (borisyankov) [inline src/settings/SettingsScreen.js]: Don't over describe the name. We don't need the 'push' part in the name.

File: src/settings/SettingsScreen.js [hunk 4]
```diff
@@ -51,22 +52,34 @@ class SettingsScreen extends React.Component {
     this.props.settingsChange('theme', value ? 'night' : 'default');
   };
 
-  handleNotificationChange = (value) => {
-    toggleOfflinePushNotifications(this.props.auth, value);
-    this.props.settingsChange('notifications', value);
+  handleOfflineNotificationChange = async (value) => {
+    await toggleMobilePushSettings({ auth: this.props.auth, offline: value });
```
Comment (borisyankov) [inline src/settings/SettingsScreen.js]: I don't think we need the `await`. This will make the toggle buttons look laggy.

**Comment by kunall17** [comment]
Done 👍

**Comment by borisyankov** [comment]
Cool. Looks done.

**Pull request merged**